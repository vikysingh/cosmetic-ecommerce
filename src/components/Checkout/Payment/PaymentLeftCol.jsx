import React from "react"
import { Link, useHistory } from "react-router-dom"
import { Breadcrumb, Input, Modal, Button } from "antd"
import routes from "../../../constants/routes.json"
import Cards from 'react-credit-cards';
import 'react-credit-cards/es/styles-compiled.css';
// import { ThemeButton } from "../../index"
import styles from "./PaymentLeftCol.module.css"

import checkouts from "../../../constants/checkouts.json"

import { useFormik } from "formik"



const { Item } = Breadcrumb

export default function PaymentLeftCol() {

    let history = useHistory()

    const [ stateFocus, setFocus ] = React.useState('')
    const [ modalVisible, setModalVisible ] = React.useState(false)

    const formik = useFormik({
        initialValues: {
            cvc: '',
            expiry: '',
            focus: '',
            name: '',
            number: '',
        },
        onSubmit: values => {
            setModalVisible(true)
        },
    })

    function handleModalClose() {
        setModalVisible(false)
        history.push(routes.cart)
    }

    return <div className={styles.PAYMENT__LEFT_COL} >
        <Breadcrumb >
            <Item>
                <Link to={routes.home}> Home </Link>
            </Item>
            <Item>
                <Link to={routes.cart}> Cart </Link>
            </Item>
            <Item>
                <Link to={routes.checkoutAddress}> Address </Link>
            </Item>
            <Item>
                Payment
            </Item>
        </Breadcrumb>

        <Modal
        title={checkouts.notices.paymentSuccessTitle}
        visible={modalVisible}
        footer={[
            <Button key="submit" type="primary" onClick={handleModalClose}>
              Ok
            </Button>
        ]}
        >
            <strong> { checkouts.notices.paymentSuccessMessage } </strong>
        </Modal>

        <Cards
          cvc={formik.values.cvc}
          expiry={formik.values.expiry}
          focused={stateFocus}
          name={formik.values.name}
          number={formik.values.number}
          preview={true}
          acceptedCards={['visa', 'mastercard']}
        />

        <form action="post" onSubmit={formik.handleSubmit} >
            <Input type="number" placeholder="Card number" name="number"
            maxLength={16} value={formik.values.number} required={true}
            onFocus={e => setFocus(e.target.name) }
            onChange={formik.handleChange} />

            <Input type="text" placeholder="Name" name="name"
            value={formik.values.name} required={true}
            onFocus={e => setFocus(e.target.name)}
            onChange={formik.handleChange} />

            <Input type="number" placeholder="Valid thru" name="expiry"
            maxLength={5} value={formik.values.expiry} required={true}
            onFocus={e => setFocus(e.target.name)}
            onChange={formik.handleChange} />

            <Input type="number" placeholder="CVC" name="cvc"
            maxLength={4} value={formik.values.cvc} required={true}
            onFocus={e => setFocus(e.target.name)}
            onChange={formik.handleChange} />

            {/* <ThemeButton theme="shop"> Complete payment </ThemeButton> */}
            <button type="submit" > Complete payment </button>
        </form>
    </div>
}