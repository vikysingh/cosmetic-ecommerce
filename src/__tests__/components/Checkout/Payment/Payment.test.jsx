import React from "react"
import { shallow } from "enzyme"

import PaymentLeftCol from "../../../../components/Checkout/Payment/PaymentLeftCol"

test('Should render Payment from Checkout', () => {
    const wrapper = shallow(<PaymentLeftCol />)

    expect(wrapper).toMatchSnapshot()
})