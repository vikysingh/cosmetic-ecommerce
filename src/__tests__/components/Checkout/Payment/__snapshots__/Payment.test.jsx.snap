// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Payment from Checkout 1`] = `
<div
  className="PAYMENT__LEFT_COL"
>
  <Breadcrumb>
    <BreadcrumbItem>
      <[object Object]
        to="/"
      >
         Home 
      </[object Object]>
    </BreadcrumbItem>
    <BreadcrumbItem>
      <[object Object]
        to="/cart"
      >
         Cart 
      </[object Object]>
    </BreadcrumbItem>
    <BreadcrumbItem>
      <[object Object]
        to="/checkout/address"
      >
         Address 
      </[object Object]>
    </BreadcrumbItem>
    <BreadcrumbItem>
      Payment
    </BreadcrumbItem>
  </Breadcrumb>
  <ReactCreditCards
    acceptedCards={
      Array [
        "visa",
        "mastercard",
      ]
    }
    cvc=""
    expiry=""
    focused=""
    locale={
      Object {
        "valid": "valid thru",
      }
    }
    name=""
    number=""
    placeholders={
      Object {
        "name": "YOUR NAME HERE",
      }
    }
    preview={true}
  />
  <form>
    <Input
      maxLength={16}
      name="number"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Card number"
      type="number"
    />
    <Input
      name="name"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Name"
      type="text"
    />
    <Input
      maxLength={5}
      name="expiry"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Valid thru"
      type="number"
    />
    <Input
      maxLength={4}
      name="cvc"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="CVC"
      type="number"
    />
    <[object Object]
      theme="shop"
    >
       Complete payment 
    </[object Object]>
  </form>
</div>
`;
